### PR TITLE
Add OIDC auth/role support

### PR DESCRIPTION
**Pull request by dwelch0**
Add logic to support creation, tuning, and deletion of oidc authentication method and oidc roles.
Update test suite to adequately test new oidc functionality

Ticket: https://issues.redhat.com/browse/APPSRE-4659

### CODE REVIEW

**Comment by dwelch0** [comment]
PR check failing because dependencies within qontract-schema and app-interface have not been merged yet

**Comment by dwelch0** [comment]
Committed tests w/ url used to validate tests locally.. Direction required on how to configure test for PR check.

**Review by TGPSKI — APPROVED** [review]

File: toplevel/role/role.go [hunk 1]
```diff
@@ -160,3 +177,51 @@ func asItems(xs []entry) (items []vault.Item) {
...
+					role.Options[k] = v
+				}
+			}
+		}
+	}
+}
+
+// remove attributes not supported in commercial but in fedramp variant
+func pruneUnsupported(roles []entry) error {
+	current, err := version.NewVersion(vault.GetVaultVersion())
```
Comment (bkez322) [inline toplevel/role/role.go:213]: `version` is a cool package

File: toplevel/role/role.go [hunk 2]
```diff
@@ -129,6 +139,9 @@ func (c config) Apply(entriesBytes []byte, dryRun bool, threadPoolSize int) {
 		}
 	}
 
+	addOptionalOidcDefaults(entries)
+	pruneUnsupported(entries)
```
Comment (bkez322) [inline toplevel/role/role.go]: I'm not a huge fan of us having to do this, but this is may very well be the least bad way of doing things.

I'd like to think that either the vault server API, or the golang vault client library, are capable of filling in these defaults itself for us. But maybe that isn't a thing.

**Review by bkez322 — APPROVED** [review]
LGTM!

File: toplevel/role/role.go (hunk 2 above)
Comment (bkez322) [reply toplevel/role/role.go]: Something to look into for the future perhaps.

**Pull request merged**